### PR TITLE
Refactor the send payload function

### PR DESCRIPTION
**Pull request by HoussemDjeghri**
This PR has 2 main changes: 
- Bind the sendBeacon function to the navigator object before passing the reference.
- Surround the sendBeacon function call with a try catch bloc to catch the url parsing errors.

This PR fixes the issue #194

### CODE REVIEW

**Comment by HoussemDjeghri** [comment]
This PR fixes the issue #194

**Comment by HoussemDjeghri** [comment]
@sarveshnagpal could you please review this PR, here is an URL example that sendBeacon couldn't parse and throw a "TypeError" or "Illegal Invocation" exception in our prod app earlier today : `/email-confirmed?result=%7B%22status%22%3Afalse%7D`

File: packages/clarity-js/src/data/upload.ts [hunk 1]
```diff
@@ -151,8 +151,14 @@ function send(payload: string, zipped: Uint8Array, sequence: number, beacon: boo
...
         if (beacon && "sendBeacon" in navigator) {
-            dispatched = navigator.sendBeacon(url, payload);
-            if (dispatched) { done(sequence); }
+            try {
+                // sendBeacon must be binded to the navigator before passing the reference
+                dispatched = navigator.sendBeacon.bind(navigator)(url, payload);
+                if (dispatched) { done(sequence); }
+            } catch (error) {
+                // sendBeacon may throw a "TypeError" exception if the url parsing fails
+                console.error(`Payload upload failed: ${error}`);
```
Comment (sarveshnagpal) [inline packages/clarity-js/src/data/upload.ts:160]: We don't log debug statements in the production code for performance reasons.

File: packages/clarity-js/src/data/upload.ts [hunk 2]
```diff
@@ -151,8 +151,14 @@ function send(payload: string, zipped: Uint8Array, sequence: number, beacon: boo
         // However, we don't want to rely on it for every payload, since we have no ability to retry if the upload failed.
         // Also, in case of sendBeacon, we do not have a way to alter HTTP headers and therefore can't send compressed payload
         if (beacon && "sendBeacon" in navigator) {
-            dispatched = navigator.sendBeacon(url, payload);
-            if (dispatched) { done(sequence); }
+            try {
+                // sendBeacon must be binded to the navigator before passing the reference
+                dispatched = navigator.sendBeacon.bind(navigator)(url, payload);
```
Comment (sarveshnagpal) [inline packages/clarity-js/src/data/upload.ts:156]: I looked up issue you referenced #194 - do you know if this is a Sentry problem? Looked up MDN: https://developer.mozilla.org/en-US/docs/Web/API/Navigator/sendBeacon and it doesn't require binding to navigator before sending sendBeacon. If you could explain more that will help.

**Review by sarveshnagpal — CHANGES_REQUESTED** [review]

File: packages/clarity-js/src/data/upload.ts (hunk 1 above)
Comment (HoussemDjeghri) [reply packages/clarity-js/src/data/upload.ts:160]: Okay @sarveshnagpal, I have removed the `console.error` call and replaced it with a "do nothing" comment

File: packages/clarity-js/src/data/upload.ts (hunk 2 above)
Comment (HoussemDjeghri) [reply packages/clarity-js/src/data/upload.ts:156]: @sarveshnagpal I couldn't find more details also on the manner, but this fix is used in popular frameworks like next.js, you can check this [commit](https://github.com/blitz-js/next.js/commit/4bb419b473394285249044c174a8ba736669f98f) from the next.js repo , they had the same issue and fixed like I did.

Comment (HoussemDjeghri) [reply packages/clarity-js/src/data/upload.ts:156]: @sarveshnagpal here is the [issue](https://github.com/vercel/next.js/issues/23856) that mansions this in the next.js repo

Comment (HoussemDjeghri) [reply packages/clarity-js/src/data/upload.ts:156]: sentry also had the same issue and fixed it like that in this [PR ](https://github.com/getsentry/sentry-javascript/pull/4008)

**Comment by HoussemDjeghri** [comment]
@sarveshnagpal could you review this PR please, we are still receiving the error in PROD.

**Review by sarveshnagpal — APPROVED** [review]

**Pull request merged**